search-service/src/main/scala/eu/jrie/put/cs/pt/scrapper/domain/results/ResultsRepository.scala

```diff
@@ -46,67 +46,98 @@ private class ResultsRepository(
 
   override def onMessage(msg: ResultsRepoMsg): Behavior[ResultsRepoMsg] = {
     msg match {
-      case AddResult(result) => addNewResult(result)
-      case FindResults(userId, searchId, taskId, replyTo) => findResults(userId, searchId, taskId, replyTo)
+      case AddResults(results) =>
+        addResults(results)
+        Behaviors.same
+      case FindResults(userId, searchId, taskId, replyTo) =>
+        findResults(userId, searchId, taskId, replyTo)
+        Behaviors.same
       case _ =>
         context.log.info("unsupported repo msg")
         getSet ! EndGetSet()
         Behaviors.stopped
     }
   }
 
-  private def addNewResult(result: Result): Behavior[ResultsRepoMsg] = {
+  private def addResults(results: Seq[Result]): Unit = {
+    context.log.debug(s"Adding ${results.length} results")
+    results.groupBy { _.taskId }
+      .foreachEntry { case (taskId, withId) =>
+        val lastIds = findLastIds(taskId)
+        val flaggedResults: Seq[Result] = Await.result(withNewcomerFlag(withId, lastIds), Duration.Inf)
```

Comment:
To jest miejce do refactoru przyszłości - działa dobrze ale prawdopodonie można to napisać lepiej